### ISSUE TITLE
Option to set ACL Rule to be the very bottom by default

### ISSUE THREAD

**Issue by btzq** [apache/cloudstack#8865]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
Improvement Request (UI, Functionality)
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Note: This ticket is broken down as requested from the origianl post, https://github.com/apache/cloudstack/issues/8841

**Option to set ACL Rule to be the very bottom by default**

- At the very end of our ACL Rule, we usually have a 'Deny All'
- But when i use API or Terraform to create ACL Rules, the arrangement and sequence (including the rule number) gets all messed up.
- Would be great if we could have a parameter and function in the UI to make the rule fix itself at the bottom of the list.

##### STEPS TO REPRODUCE

N/A

~~~
N/A
~~~

##### EXPECTED RESULTS

~~~
To be able to manage ACL Rules more easily for large scale projects
~~~

##### ACTUAL RESULTS

~~~
Challenging to use ACL rules for large scale projects.
~~~

**Comment by DaanHoogland** [comment]
@btzq , you can use the parameter `number` to set the order of an acl item in the acl list. Does this fullfill this functionality? We may need to improve the documentation for this to make it more clear.

**Comment by DaanHoogland** [comment]
I think the createNetworkAcl should be renamed to createNetworkAclItem to be more consistent with the other commands. Same for deleteNetworkACL(Item)!

**Comment by btzq** [comment]
@DaanHoogland i wanst sure that the 'number' parameter was working. Because i was intuitively expecting that the row would automatically be moved/sorted to the bottom of the page. I think i was expecting this as well cause the ACL Rule row can be dragged up and down manually. So when i saw that the rule wasnt being moved to the bottom (when i keyed in the number parameter), i assumed it wasnt working.

**Comment by DaanHoogland** [comment]
> @DaanHoogland i wanst sure that the 'number' parameter was working. Because i was intuitively expecting that the row would automatically be moved/sorted to the bottom of the page. I think i was expecting this as well cause the ACL Rule row can be dragged up and down manually. So when i saw that the rule wasnt being moved to the bottom (when i keyed in the number parameter), i assumed it wasnt working.

That's alright @btzq , but does this satisfy this request/issue?

**Comment by btzq** [comment]
I think at least, the Cloudstack GUI should auto sort the ACL Rules based on the 'number'. 

It would just be a UI change and would improve the usability.

**Comment by DaanHoogland** [comment]
> I think at least, the Cloudstack GUI should auto sort the ACL Rules based on the 'number'.

I think it does, at least from the UI. Can you describe the scenario in which it doesn't?

**Comment by btzq** [comment]
@DaanHoogland i just retested and i think i know the confusion:

- Lets say i have 10 ACL Rules in the List. All numbered from 1-10.
- If i change Rule 2 manually to be 'Rule Number = 200', then i can see the rule being sorted automatically to the bottom. 
- But now, if i look back at the top of the rule list, i see now the rules in the order of (1,3,4,5,6,7,8,9,10,200)

And if you were to try to rearange things later on, it becomes super complicated because it doesnt re-initiliase the rule numbers. For example:

- Lets say we have (1,3,4,5,6,7,8,9,10,200)
- If you try to drag 4 to the position of 2 (in between 1 and 3), the result you get is the rule number 3 (when it should be 2)

So maybe what we need instead to make it less confusing is to re-initailise the rule numbers?

**Comment by DaanHoogland** [comment]
ok, makes sense. This could also be a UI implementation or as alternative an extra API `reorderAclList` or the likes. I would expect it to get the id of the list and an `interval` option, in order to specify that instead of 1,2,3,4,5,6,7,8,9,10 I want 1,3,5,7,9,11,13,15,17,19 as a result, or 10,20,30,40,50,60,70,80,90,100.

**Comment by shwstppr** [comment]
Fixed with #9119 

Adding a new API for re-ordering list with rule numbers can be a separate discussion
